webapp/store/index.js

```diff
@@ -0,0 +1,37 @@
+import login from "../gql/login.gql";
+import jwt_decode from "jwt-decode";
+
+export const state = () => ({
+    token: null,
+    currentUser: null,
+});
+
+export const getters = {
+    loggedIn(state) {
+        return !!state.token;
+    },
+};
+
+export const mutations = {
+    setToken(state, token) {
+        state.token = token;
+        if (token) {
+            let { userId } = jwt_decode(token);
+            state.currentUser = userId;
+        } else {
+            state.currentUser = null;
+        }
```

Comment:
Maybe consider to only set the token if it is valid. Otherwise (if the token contains garbage) your state will be broken because your `loggedIn` method checks `!!state.token`. So it will show True even if the token is invalid.